src/utilities.jl

```diff
@@ -240,27 +242,54 @@ function Base.show(io::IO, pca::PCA;
                 pca.corr ? "correlation" : "(relative) covariance",
                 " matrix")
         # only display the lower triangle of symmetric matrix
-        Base.print_matrix(io, round.(LowerTriangular(pca.covcor), digits=ndigitsmat))
+        printmat = round.(LowerTriangular(pca.covcor), digits=ndigitsmat)
+        if pca.rnames !== missing 
+            n = length(pca.rnames)
+            cv = string.(printmat)
+            dotpad = lpad(".", div(maximum(length, cv),2))
+            for i = 1:n, j = (i+1):n
+                cv[i, j] = dotpad
+            end
+            neg = startswith.(cv, "-")
+            if any(neg)
+                cv[.!neg] .= " ".* cv[.!neg]
+            end
```

Comment:
```suggestion
            for i = 1:length(cv)
                if !startswith(cv[i], "-")
                    cv[i] = " " * cv[i]
                end
            end
```
Slightly longer but it has the benefit of only making one pass over `cv`. Otherwise you're doing one to broadcast `startswith`, one to check `any`, and then one for the broadcasted indexed assignment.

Reply:
But this will always add the space prefix for positive numbers, even if there are no negative numbers, right? I guess the extra space really doesn't matter for printing because everything remains aligned.

Reply:
I agree with Phillip that the check for any negative numbers is probably needed.  Could the check be written as
```
if any(startswith("-"), cv)
```
so that it dispatches to the `any(f::Function, a::AbstractArray; dims) in Base at reducedim.jl:732` method? I think that would avoid allocation of the intermediate result.

Reply:
I guess my scheme doesn't work out that well because you then have to do the `startswith(cv[i], "-")` check again to add the spaces.

It seems that in @palday 's version `neg` will be a BitArray and cv vector is usually short so I don't think there will be much overhead in allocating `neg`.